includes/forms/class-llms-forms.php

```diff
@@ -630,6 +630,85 @@ public function get_form_post( $location, $args = array() ) {
 
 	}
 
+	/**
+	 * Check whether a given form is a core form.
+	 *
+	 * When in presence of more than one form per location, the core form is identified as the one with lower id.
+	 *
+	 * @since [version]
+	 *
+	 * @param WP_Post|int $form Form's WP_Post instance, or its ID.
+	 * @return boolean
+	 */
+	public function is_a_core_form( $form ) {
+
+		$form_id = $form instanceof WP_Post ? $form->ID : $form;
+
+		if ( ! $form_id ) {
+			return false;
+		}
+
+		return in_array( $form_id, $this->get_core_forms( 'ids' ), true );
+
+	}
+
+	/**
+	 * Retrieves only core forms.
+	 *
+	 * When in presence of more than one form per location it selects the one with lower id.
+	 *
+	 * @since [version]
+	 *
+	 * @param string $return What to return: 'posts', for an array of WP_Post; 'ids' for an array of WP_Post ids.
+	 * @return WP_Post[]|int[]
+	 */
+	private function get_core_forms( $return = 'posts', $use_cache = true ) {
+
+		global $wpdb;
+
+		$forms_cache_key = 'posts' === $return ? 'llms_core_forms' : 'llms_core_form_ids';
+		$forms           = $use_cache ? wp_cache_get( $forms_cache_key ) : false;
+
+		if ( false !== $forms ) {
+			return $forms;
+		}
+
+		$locations              = array_keys( $this->get_locations() );
+		$locations_placeholders = implode( ',', array_fill( 0, count( $locations ), '%s' ) );
+		$prepare_values         = array_merge( $locations, array( $this->get_post_type() ) );
+
+		$query = "
+SELECT MIN({$wpdb->posts}.ID) AS ID
+FROM $wpdb->posts
+INNER JOIN {$wpdb->postmeta} ON ( {$wpdb->posts}.ID = {$wpdb->postmeta}.post_id )
+INNER JOIN {$wpdb->postmeta} AS mt1 ON ( {$wpdb->posts}.ID = mt1.post_id )
+WHERE 1=1
+AND (
+	( {$wpdb->postmeta}.meta_key='_llms_form_location' AND {$wpdb->postmeta}.meta_value IN ({$locations_placeholders}) )
+	AND
+	( mt1.meta_key='_llms_form_is_core' AND mt1.meta_value='yes' )
+	)
+AND {$wpdb->posts}.post_type=%s
+GROUP BY {$wpdb->postmeta}.meta_value";
```

Comment:
Improve table aliases. Move `WHERE` table conditions to `JOIN` clauses. Remove unnecessary parenthesis and `1=1` condition.
```suggestion
		$prepare_values         = array_merge( array( $this->get_post_type() ), $locations );

		$query = "
SELECT MIN({$wpdb->posts}.ID) AS ID
FROM $wpdb->posts
INNER JOIN {$wpdb->postmeta} AS locations ON {$wpdb->posts}.ID = locations.post_id AND locations.meta_key='_llms_form_location'
INNER JOIN {$wpdb->postmeta} AS is_cores ON {$wpdb->posts}.ID = is_cores.post_id AND is_cores.meta_key='_llms_form_is_core'
WHERE {$wpdb->posts}.post_type = %s
AND locations.meta_value IN ({$locations_placeholders})
AND is_cores.meta_value = 'yes'
GROUP BY locations.meta_value";
```

Reply:
I like what you suggest! <3
I was using WP_Query at first, + filters on the select and group clauses, so I just "copied" the SQL produced by wp, hence that `AND 1=1`  :D